instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/url/UrlAttributesGetter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.url;
+
+import javax.annotation.Nullable;
+
+/**
+ * An interface for getting URL attributes.
+ *
+ * <p>Instrumentation authors will create implementations of this interface for their specific
+ * library/framework. It will be used by the {@link UrlAttributesExtractor} (or other convention
+ * specific extractors) to obtain the various URL attributes in a type-generic way.
+ */
+public interface UrlAttributesGetter<REQUEST> {
+
+  /**
+   * Returns the absolute URL describing a network resource according to <a
+   * href="https://www.rfc-editor.org/rfc/rfc3986">RFC3986</a>.
+   *
+   * <p>Examples: {@code https://www.foo.bar/search?q=OpenTelemetry#SemConv}; {@code //localhost}
+   */
+  @Nullable
+  default String getFullUrl(REQUEST request) {
+    return null;
+  }
+
+  /**
+   * Returns the <a href="https://www.rfc-editor.org/rfc/rfc3986#section-3.1">URI scheme</a>
+   * component identifying the used protocol.
+   *
+   * <p>Examples: {@code https}, {@code ftp}, {@code telnet}
+   */
+  @Nullable
+  default String getUrlScheme(REQUEST request) {
+    return null;
+  }
```

Comment:
I think I slightly prefer consistency with other `*Getter` classes of not repeating `Url` in the method names, e.g.
* `getFull()`
* `getScheme()`
* etc

Reply:
That was my initial idea to. 

But, the problem with the HTTP semconv is, it uses other semconv (URL, client, server, network) and slightly overrides how some of the "inherited" attributes are extracted. For example, `client.address` will be extracted from the `Forwarded` header if not provided; or `server.port` will be taken from the `Host` header, but only if it's not 80/443. 

To apply these customizations we either have to pass multiple getters to the extractor (which is what we're doing right now; in the new semconv it would be 5 getters instead of 2 though), or have the HTTP getter extend the "inherited" semconv getters - in which case, something like`getFull()` appearing on an HTTP getter would be mildly confusing to the user. Which is why I've chosen to be a bit more verbose here.